psycop/common/model_training_v2/trainer/task/estimator_steps/xgboost.py

```diff
@@ -22,3 +33,33 @@ def xgboost_classifier_step(
             n_estimators=n_estimators, max_depth=max_depth, tree_method=tree_method, missing=np.nan
         ),
     )
+
+
+@BaselineRegistry.estimator_steps_suggesters.register("xgboost_suggester")
+class XGBoostSuggester(Suggester):
+    def __init__(
+        self,
+        n_estimators: IntegerspaceT = (100, 1200, True),
+        alpha: FloatSpaceT = (1e-8, 0.1, True),
+        lmbda: FloatSpaceT = (1e-8, 1.0, True),
+        max_depth: IntegerspaceT = (1, 10, True),
+        learning_rate: FloatSpaceT = (1e-8, 1, True),
+        gamma: FloatSpaceT = (1e-8, 0.001, True),
+        grow_policy: CategoricalSpaceT = ("depthwise", "lossguide"),
+    ):
+        self.n_estimators = IntegerSpace.from_list_or_mapping(n_estimators)
+        self.alpha = FloatSpace.from_list_or_mapping(alpha)
+        self.lmbda = FloatSpace.from_list_or_mapping(lmbda)
+        self.max_depth = IntegerSpace.from_list_or_mapping(max_depth)
+        self.learning_rate = FloatSpace.from_list_or_mapping(learning_rate)
+        self.gamma = FloatSpace.from_list_or_mapping(gamma)
```

Comment:
This is a little annoying. We could probably auto-generate it by introspecting the annotations, but maybe not worth it yet. 

Perhaps something to add to scikit-lightning.